File: 1094/1094.go
package main

import "fmt"

func main() {
	t := [][]int{{2, 1, 5}, {3, 3, 7}}
	r := carPooling(t, 4)
	fmt.Println(r)
}

func carPooling(trips [][]int, capacity int) bool {
	// m := make(map[int]int, 1001) // map写法
	m := [1001]int{0} // 更快一些
	for i := len(trips) - 1; i >= 0; i-- {
		m[trips[i][1]] += trips[i][0]
		m[trips[i][2]] -= trips[i][0]
	}
	p := 0
	for i := 0; i <= 1000; i++ {
		p += m[i]
		if p > capacity {
			return false
		}
	}
	return true
}

File: 1094/1094.cpp
#include "../leetcode.h"    

class Solution {
public:
    bool carPooling(vector<vector<int>>& trips, int capacity) {
        int road[1001] = {0};
        int n = trips.size();
        for (int i=0; i<n; i++){
            road[trips[i][1]] += trips[i][0];
            road[trips[i][2]] -= trips[i][0];
        }
        int psg = 0;
        for (int i = 0; i<=1000; i++){
            if (road[i] != 0){
                psg += road[i];
                if (psg > capacity) return false;
            }
        }
        return true;
    }
};